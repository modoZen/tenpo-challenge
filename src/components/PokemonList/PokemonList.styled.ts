import styled from 'styled-components';

export const PokemonListContainer = styled.div`
  margin: 0 auto;
  padding: 1.25rem 1rem;
  max-width: 1700px;

  display: grid;
  grid-template-columns: repeat(auto-fill, 15.625rem);
  column-gap: 1.25rem;
  row-gap: 2rem;
  justify-content: center;

  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
  transition-property: all;
  transition-duration: 0.5s;
`;
