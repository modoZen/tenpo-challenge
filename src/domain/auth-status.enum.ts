export enum AuthStatus {
  Checking = 'checking',
  NotAuthenticated = 'not-authenticated',
  Authenticated = 'authenticated',
}
