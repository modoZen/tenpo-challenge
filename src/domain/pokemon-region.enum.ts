export enum PokemonRegion {
  All = 'All',
  Kanto = 'Kanto',
  Johto = 'Johto',
  Hoenn = 'Hoenn',
  Sinnoh = 'Sinnoh',
  Unova = 'Unova',
  Kalos = 'Kalos',
  Alola = 'Alola',
  Galar = 'Galar',
  Hisui = 'Hisui',
  Paldea = 'Paldea',
}
