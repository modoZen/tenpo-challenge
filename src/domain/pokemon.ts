import { PokemonRegion } from './pokemon-region.enum';

export interface Pokemon {
  name: string;
  image: string;
  id: number;
  url: string;
  region: PokemonRegion;
}
