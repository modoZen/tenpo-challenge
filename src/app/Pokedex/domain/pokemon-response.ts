import { Pokemon } from './pokemon';

export interface PokemonResponse {
  count: number;
  next: string;
  previous: string;
  results: Pick<Pokemon, 'name' | 'url'>[];
}
