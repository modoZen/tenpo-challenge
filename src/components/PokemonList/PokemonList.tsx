import { type FC } from 'react';
import { PokemonListContainer } from './PokemonList.styled';
import { PokemonCard } from '../PokemonCard/PokemonCard';
import { Pokemon } from '../../domain/pokemon';

interface PokemonListProps {
  pokemons: Pokemon[];
}

export const PokemonList: FC<PokemonListProps> = ({ pokemons }) => {
  if (pokemons.length === 0) {
    return <div style={{ minHeight: '80vh' }} />;
  }

  return (
    <PokemonListContainer>
      {pokemons.map((pokemon) => (
        <PokemonCard key={pokemon.id} pokemon={pokemon} />
      ))}
    </PokemonListContainer>
  );
};
