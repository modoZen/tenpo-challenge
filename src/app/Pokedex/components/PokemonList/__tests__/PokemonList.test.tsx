import { render, screen } from '@testing-library/react';
import { Pokemon } from '../../../domain/pokemon';
import { PokemonList } from '../PokemonList';
import { pokemonListStub } from '../../../__stubs__/pokemonList.stub';

jest.mock('../../PokemonCard/PokemonCard', () => ({
  PokemonCard: ({ pokemon: { name } }: { pokemon: Pokemon }) => (
    <div data-testid='pokemon-card'>{name}</div>
  ),
}));

describe('PokemonList', () => {
  it('matches snapshot with pokemons', () => {
    const { asFragment } = render(<PokemonList pokemons={pokemonListStub} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot when empty', () => {
    const { asFragment } = render(<PokemonList pokemons={[]} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correct number of pokemon cards', () => {
    render(<PokemonList pokemons={pokemonListStub} />);
    const cards = screen.getAllByTestId('pokemon-card');
    expect(cards).toHaveLength(pokemonListStub.length);
  });
});
