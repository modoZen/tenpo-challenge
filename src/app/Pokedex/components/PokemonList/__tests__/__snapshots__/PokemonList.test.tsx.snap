// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonList matches snapshot when empty 1`] = `
<DocumentFragment>
  <div
    style="min-height: 80vh;"
  />
</DocumentFragment>
`;

exports[`PokemonList matches snapshot with pokemons 1`] = `
<DocumentFragment>
  .c0 {
  margin: 0 auto;
  padding: 1.25rem 1rem;
  max-width: 1700px;
  display: grid;
  grid-template-columns: repeat(auto-fill, 15.625rem);
  column-gap: 1.25rem;
  row-gap: 2rem;
  justify-content: center;
  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
  transition-property: all;
  transition-duration: 0.5s;
}

<div
    class="c0"
  >
    <div
      data-testid="pokemon-card"
    >
      Pikachu
    </div>
    <div
      data-testid="pokemon-card"
    >
      Dragonite
    </div>
  </div>
</DocumentFragment>
`;
